src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -283,6 +293,51 @@ public void InitializeService(ServiceHost serviceHost, SqlToolsContext context)
 
         #region Request Handlers
 
+        /// <summary>
+        /// Completion extension load request callback
+        /// </summary>
+        /// <param name="param"></param>
+        /// <param name="requestContext"></param>
+        /// <returns></returns>
+        internal async Task HandleCompletionExtLoadRequest(CompletionExtensionParams param, RequestContext<bool> requestContext)
+        {
+            try
+            {
+                var serviceProvider = (ExtensionServiceProvider)ServiceHostInstance.ServiceProvider;
+                serviceProvider.AddAssembliesToConfiguration(new Assembly[] { AssemblyLoadContext.Default.LoadFromAssemblyPath(param.Assembly) });
+                foreach (var provider in serviceProvider.GetServices<ICompletionExtensionProvider>())
+                {
+                    var cancellationTokenSource = new CancellationTokenSource();
+                    cancellationTokenSource.CancelAfter(ExtensionLoadingTimeout);
+                    var cancellationToken = cancellationTokenSource.Token;
+                    var ext = await provider.CreateAsync(param.Properties ?? new Dictionary<string, object>(), cancellationToken).WithTimeout(ExtensionLoadingTimeout);
+                    if (ext == null)
+                    {
+                        await requestContext.SendError("Failed to create ICompletionExtension");
+                        return;
+                    }
+                    string extName = ext.Name;
+                    await ext.Initialize(cancellationToken).WithTimeout(ExtensionLoadingTimeout);
+                    cancellationTokenSource.Dispose();
+                    if (!string.IsNullOrEmpty(extName))
+                    {
+                        _completionExtensions.AddOrUpdate(extName, ext, (_, previous) =>
+                        {
+                            previous?.Dispose();
+                            return ext;
+                        });
+                    }
+                    await requestContext.SendResult(true);
```

Comment:
This needs to be outside the for loop. Otherwise if you do have 2 or more providers you'll send twice.

Reply:
Good catch, fixed now.